internal/rest/dequeue_nodes.go

```diff
@@ -165,7 +165,10 @@ func (rest *RestOperations) CheckAndPublishForRetry(err error, publishKey, key s
 	if webSyncErr, ok := err.(*utils.WebSyncError); ok {
 		aviError, ok := webSyncErr.GetWebAPIError().(session.AviError)
 		if ok && aviError.HttpStatusCode == 401 {
-			if avimodel != nil && avimodel.GetRetryCounter() != 0 {
+			if strings.Contains(*aviError.Message, "Invalid credentials") {
```

Comment:
we have to rely on strings ? is that what we get even if session expires ? Is there something similar even if certificate is changed ?

Reply:
We had this discussion and I was in the middle drafting a mail to the controller folks. We are relying on such strings at multiple places. There's no good way to use constants here at the moment. 401 is used for multiple things, one more example:
```401; error from Avi: map[detail:Authentication credentials were not provided.]```